src/snappy_device_agents/devices/maas2/maas2.py

```diff
@@ -223,9 +229,41 @@ def _run_tpm_clear_cmd(self):
             return True
         return False
 
-    def deploy_node(self, distro="bionic", kernel=None, user_data=None):
+    def deploy_node(
+        self, distro="bionic", kernel=None, user_data=None, storage_data=None
+    ):
         # Deploy the node in maas, default to bionic if nothing is specified
         self.recover()
+        status = self.node_status()
+        # configuring storage must take place when node is in a ready state
+        if storage_data:
```

Comment:
Note - used this evaluation as this protects against proceeding in the instance of '{'disks': {}}'
vs. evaluating explicitly against None.

Reply:
worth adding that as a comment in the code too